### PR TITLE
Add unstructured converter and test fixtures

### PR DESCRIPTION
**Pull request by lblackstone**

### CODE REVIEW

File: pkg/await/fixtures/deployment.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+							Image: "nginx",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func DeploymentBasic_Uns() *unstructured.Unstructured {
```
Comment (metral) [inline pkg/await/fixtures/deployment.go]: nit: can we drop the `_` and spell out `Unstructured` - might as well be explicit and help readability where possible

File: pkg/clients/unstructured.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+)
+
+func FromUnstructured(obj *unstructured.Unstructured) (metav1.Object, error) {
+	b, err := obj.MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+
+	var output metav1.Object
+	switch kinds.Kind(obj.GetKind()) {
```
Comment (metral) [inline pkg/clients/unstructured.go]: Can we have docs stating which objects are supported today vs not?

File: pkg/clients/unstructured_test.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import (
+	"reflect"
+	"testing"
+
+	"github.com/pulumi/pulumi-kubernetes/pkg/await/fixtures"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func TestFromUnstructured(t *testing.T) {
```
Comment (metral) [inline pkg/clients/unstructured_test.go]: Are there plans to test the rest of the supported objects other than the Pod and Deployment?

Comment (lblackstone) [reply pkg/clients/unstructured_test.go]: Yeah, probably should have been more clear in the description, but this PR is basically a prototype to make sure the approach made sense before I fully committed to it. My intent is to expand these fixtures to replace all the giant YAML strings in the unit tests. I'll keep fleshing things out over multiple PRs.

File: pkg/clients/unstructured.go (hunk 2 above)
Comment (lblackstone) [reply pkg/clients/unstructured.go]: Yeah, I'll add docs eventually, but for right now, this is the set of objects we're using the in complex await logic.

**Pull request merged**